django/db/backends/mysql/operations.py

```diff
@@ -142,6 +143,8 @@ def adapt_datetimefield_value(self, value):
             return None
 
         # MySQL doesn't support tz-aware datetimes
+        if isinstance(value, Combinable):
```

Comment:
If the immediately following clause were really the _only_ reason the short-circuit was required, then I'd say don't short-circuit, just include the combinable check in a compound conditional for that clause. That way you avoid the possibility of accidentally short-circuiting other code that might later be added later in the method, that shouldn't be short-circuited.

But: I don't think it's accurate that the following clause is the only thing that can't handle a Combinable -- the `supports_microsecond_precision` check following it also should be skipped. In general it seems likely to me that the remainder of this method will always assume it has a datetime, so a short-circuit return is correct. So I think the comment should stay with the clause it applies to.